packages/terra-embedded-component/src/EmbeddedComponent.jsx

```diff
@@ -1,31 +1,146 @@
+/* global SystemJS */
 import React from 'react';
 import PropTypes from 'prop-types';
-import classNames from 'classnames/bind';
-import 'terra-base/lib/baseStyles';
-import styles from './EmbeddedComponent.scss';
+import axios from 'axios';
+import 'systemjs/dist/system';
+import { loadStylesheet } from './assets';
 
-const cx = classNames.bind(styles);
+// Expose AMD required globals on the window
+window.define = window.SystemJS.amdDefine;
+window.require = window.requirejs = SystemJS.amdRequire;
+
+const DefaultTimeout = () => <p>Component mounting is timed out.</p>;
 
 const propTypes = {
- /*
- * Content to be displayed as the name
- */
-  name: PropTypes.string,
+
+  /**
+   * The basePath to load the manifest and manifest asset dependencies from.
+   * */
+  basePath: PropTypes.string.isRequired,
+
+  /**
+   * A placeholder to display while the component is loading.
+   * */
+  placeholder: PropTypes.element,
+
+  /**
+   * A component to display while the loading of css or module is timed out.
+   * */
+  timeoutComponent: PropTypes.element,
+
+  /**
+   * The number of milliseconds to wait before timing out.
+   * @type {[type]}
+   */
+  timeoutInterval: PropTypes.number,
+
+  /**
+   * The entry file in the manifest to use to load the embeddable module.
+   * */
+  entry: PropTypes.string,
+
+  /**
+   * A manifest object defining the asset dependencies. If provided will be used instead
+   * of fetching a manifest from the basePath.
+   * */
+  manifest: PropTypes.object,
 };
 
 const defaultProps = {
-  name: 'default',
+  entry: 'index.js',
+  timeoutInterval: 3000,
```

Comment:
there isn't really an "interval", it would just be timeout.

Reply:
Also, 3 seconds is really low, Bump this up to like 10s or something

Reply:
By default the component shouldn't time out.... What is an acceptable timeout will range greatly by what is being loaded, where it is hosted, what type of network the user can be expected to be using, and the type of advice.. as well as teh applicaiton loading it (browsers can only load so many assets at once).

Reply:
With the latest commit, the component only times out it a timeout is given. The default 'timeout' prop is now -1, and the component will only time out if the prop is set to greater than 0.